src/factories/createPoolConfiguration.ts

```diff
@@ -58,6 +58,16 @@ export const createPoolConfiguration = (dsn: string, clientConfiguration: Client
     }
   }
 
+  if (clientConfiguration.idleTimeout !== 'DISABLE_TIMEOUT') {
+    if (clientConfiguration.idleTimeout === 0) {
+      log.warn('idleTimeout=0 sets timeout to 0 milliseconds; use idleTimeout=DISABLE_TIMEOUT to disable timeout');
+
+      poolConfiguration.idleTimeoutMillis = 1;
+    } else {
+      poolConfiguration.idleTimeoutMillis = clientConfiguration.idleTimeout;
+    }
+  }
+
```

Comment:
If `idleTimeoutMillis` option is not set, `pg-pool` sets this to 10,000 milliseconds.
https://github.com/brianc/node-postgres/blob/a4ef6ce38c1e04bad2215312b1c79e64654cc857/packages/pg-pool/index.js#L92

If `clientConfiguration.idleTimeout` is set to `DISABLE_TIMEOUT`, then `poolConfiguration.idleTimeoutMillis` should be set to 0 to disable auto-disconnection of idle clients.